subsys/dfu/fmfu_fdev/cddl/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Output directories inside build dir
+set(src_out ${ZEPHYR_BINARY_DIR}/source/generated)
+set(include_out ${ZEPHYR_BINARY_DIR}/include/generated)
+
+# Make sure that output directory for *.c files exist
+file(MAKE_DIRECTORY ${src_out})
+
+# This file is used as the source for the parser generator
+set(cddl_file ${CMAKE_CURRENT_LIST_DIR}/modem_update.cddl)
+
+# These are the entry types needed by the source code
+set(entry_types Wrapper Sig_structure1 Segments)
+
+set(decode_c_name modem_update_decode.c)
+set(decode_h_name modem_update_decode.h)
+set(types_h_name modem_update_types.h)
+set(decode_c ${src_out}/${decode_c_name})
+set(decode_h ${include_out}/${decode_h_name})
+set(types_h ${include_out}/${types_h_name})
+set(src_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/src)
+set(include_install_dir ${NRF_DIR}/subsys/dfu/fmfu_fdev/include)
+set(license ${CMAKE_CURRENT_LIST_DIR}/license.txt)
+
+function(cat IN_FILE OUT_FILE)
+  file(READ ${IN_FILE} CONTENTS)
+  file(APPEND ${OUT_FILE} "${CONTENTS}")
+endfunction()
+
+foreach(target decode_c decode_h types_h)
+  list(APPEND CAT_COMMANDS COMMAND cat ${license} ${${target}} > ${${target}})
+  list(APPEND CF_COMMANDS COMMAND clang-format -i ${${target}})
+endforeach()
+
+add_custom_command(
+  OUTPUT ${decode_c} ${decode_h} ${types_h}
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_CDDL_GEN_MODULE_DIR}/scripts/cddl_gen.py
+  -i ${cddl_file}
+  --oc ${decode_c}
+  --oh ${decode_h}
+  --oht ${types_h}
+  -t ${entry_types}
+  -d # Decode
+  --default-maxq 16 # Up to 16 segments is supported.
+  ${CAT_COMMANDS}
+  ${CF_COMMANDS}
+  COMMENT
+  "Re-generating files based on ${cddl_file}"
+  DEPENDS ${license}
+  )
+
+zephyr_library()
+zephyr_library_sources(${decode_c})
+zephyr_include_directories(${include_out})
+
+# Create install target which allows the user to 'install' the generated
+# parser files into the working tree.
+list(APPEND INSTALL_COMMANDS
+  COMMAND ${CMAKE_COMMAND} -E echo "Installing Modem Update CDDL decoder files"
```

Comment:
Please remove.

If you want messages printed during the target execution, use `COMMENT` on the `add_custom_target()` instead.

```suggestion
```